File: Android/app/src/main/java/com/edu/primary/database/dao/WrongQuestionDao.java
package com.edu.primary.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import com.edu.primary.database.entity.WrongQuestionEntity;

import java.util.List;

@Dao
public interface WrongQuestionDao {
    @Insert
    long insertWrongQuestion(WrongQuestionEntity wrongQuestion);

    @Update
    void updateWrongQuestion(WrongQuestionEntity wrongQuestion);

    @Delete
    void deleteWrongQuestion(WrongQuestionEntity wrongQuestion);

    @Query("SELECT * FROM wrong_questions WHERE userId = :userId ORDER BY wrongTime DESC")
    List<WrongQuestionEntity> getWrongQuestionsByUser(long userId);

    @Query("SELECT * FROM wrong_questions WHERE userId = :userId AND questionId = :questionId")
    WrongQuestionEntity getWrongQuestion(long userId, long questionId);
}

File: Android-1/app/src/main/java/com/edu/primary/data/local/dao/WrongQuestionDao.kt
package com.edu.primary.data.local.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.edu.primary.data.local.entity.WrongQuestionEntity

@Dao
interface WrongQuestionDao {
    @Insert
    suspend fun insert(wrong: WrongQuestionEntity): Long

    @Update
    suspend fun update(wrong: WrongQuestionEntity)

    @Delete
    suspend fun delete(wrong: WrongQuestionEntity)

    @Query("SELECT * FROM wrong_questions WHERE userId = :userId ORDER BY wrongTime DESC")
    suspend fun getByUser(userId: Long): List<WrongQuestionEntity>

    @Query("SELECT * FROM wrong_questions WHERE userId = :userId AND questionId = :questionId")
    suspend fun get(userId: Long, questionId: Long): WrongQuestionEntity?
}
